webapp/src/toolbox.tsx

```diff
@@ -774,57 +752,29 @@ export class TreeRow extends data.Component<TreeRowProps, {}> {
             && appTheme.blocklyOptions.toolboxOptions.invertedMultiplier || 0.3;
 
         let treeRowStyle: React.CSSProperties = {
-            paddingLeft: '0px'
-        }
-        let treeRowClass = 'blocklyTreeRow';
-        if (appTheme.invertedToolbox) {
-            // Inverted toolbox
-            treeRowStyle.backgroundColor = (metaColor || '#ddd');
-            treeRowStyle.color = '#fff';
-        } else {
-            if (appTheme.coloredToolbox) {
-                // Colored toolbox
-                treeRowStyle.color = `${metaColor}`;
-            }
-            const border = `8px solid ${metaColor}`;
-            if (isRtl) {
-                treeRowStyle.borderRight = border;
-            } else {
-                treeRowStyle.borderLeft = border;
-            }
-            if (topRowIndex && this.props.shouldAnimate) {
-                treeRowStyle.animationDelay = `${(topRowIndex * this.animationDelay) + this.baseAnimationDelay}s`;
-                treeRowClass += ' blocklyTreeAnimate';
-            }
-        }
+            paddingLeft: '0px',
+            "--block-meta-color": metaColor,
+            "--block-faded-color": pxt.toolbox.fadeColor(metaColor || '#ddd', invertedMultipler, false)
+        } as React.CSSProperties;
 
-        // Selected
-        if (selected) {
-            treeRowClass += ' blocklyTreeSelected';
-            if (appTheme.invertedToolbox) {
-                treeRowStyle.backgroundColor = `${pxt.toolbox.fadeColor(metaColor, invertedMultipler, false)}`;
-            } else {
-                treeRowStyle.backgroundColor = (metaColor || '#ddd');
-            }
-            treeRowStyle.color = '#fff';
+        let treeRowClass = `blocklyTreeRow${selected ? ' blocklyTreeSelected' : '' }`;
+
+        if (topRowIndex && this.props.shouldAnimate) {
+            treeRowStyle.animationDelay = `${(topRowIndex * this.animationDelay) + this.baseAnimationDelay}s`;
+            treeRowClass += ' blocklyTreeAnimate';
         }
 
         // Icon
-        const iconClass = `blocklyTreeIcon${subns ? 'more' : icon ? (nameid || icon).toLowerCase() : 'Default'}`.replace(/\s/g, '');
+        let iconClass = `blocklyTreeIcon${subns ? 'more' : icon ? (nameid || icon).toLowerCase() : 'Default'}`.replace(/\s/g, '');
         let iconContent = subns ? pxt.toolbox.getNamespaceIcon('more') : icon || pxt.toolbox.getNamespaceIcon('default');
-        let iconImageStyle: JSX.Element;
+        let iconImageStyle: React.CSSProperties = {
+            "--image-icon-url": `url("${Util.pathJoin(pxt.webConfig.commitCdnUrl, encodeURI(icon))}")!important`,
```

Comment:
```suggestion
            "--image-icon-url": iconContent.length > 1 ? `url("${Util.pathJoin(pxt.webConfig.commitCdnUrl, encodeURI(icon))}")!important` : undefined,
```

nit: might be worth only setting this when `iconContent.length > 1` just for future debugging sake, since as written the variable will be set even when it won't be used / doesn't actually point to a valid url. Wouldn't cause a behavioral difference though, just to prevent people from getting weirded out on why they're seeing seemingly invalid urls in css. Can hoist that `iconContent.length > 1` out to a `const isImageIcon = iconContent.length > 1` as well if you want to dedup / simplify a little more